tests/test_tabulation.py

```diff
@@ -120,3 +120,332 @@ def runTest(self):
         self.assertEqual(boxcar.square_width(), 99.)
+
+        # Not 1 dimensional x
+        x = np.array([[1, 2], [3, 4]])  # 2-dimensional array
+        y = np.array([4, 5])
+        with self.assertRaises(ValueError) as context:
+            Tabulation(x, y)
+        self.assertEqual(str(context.exception), "x array in not 1-dimensional")
+
+        # Test initialization with x and y arrays of different sizes
+        x = np.array([1, 2, 3])
+        y = np.array([4, 5])  # Mismatched size
+        with self.assertRaises(ValueError) as context:
+            Tabulation(x, y)
+        self.assertEqual(str(context.exception), "x and y arrays do not have the same size")
+
+        # Test initialization with a non-monotonic x array
+        x = np.array([1, 3, 2])  # Non-monotonic
+        y = np.array([4, 5, 6])
+        with self.assertRaises(ValueError) as context:
+            Tabulation(x, y)
+        self.assertEqual(str(context.exception), "x-coordinates are not monotonic")
+
+        # Test update with new_y set to None
+        x = np.array([1, 2, 3])
+        y = np.array([4, 5, 6])
+        tab = Tabulation(x, y)
+        result = tab._update_y(None)
+        self.assertIs(result, tab)  # Should return the original object
+
+        # Test update with new_y having a different size than x
+        x = np.array([1, 2, 3])
+        y = np.array([4, 5, 6])
+        tab = Tabulation(x, y)
+        new_y = np.array([7, 8])  # Mismatched size
+        with self.assertRaises(ValueError) as context:
+            tab._update_y(new_y)
+        self.assertEqual(str(context.exception), "x and y arrays do not have the same size")
+
+        # Test xmerge with non-overlapping domains
+        x1 = np.array([1, 2, 3])
+        x2 = np.array([4, 5, 6])
+        with self.assertRaises(ValueError) as context:
+            result = Tabulation._xmerge(x1, x2)
+        self.assertEqual(str(context.exception), "domains do not overlap")
+
+        # resample where x=None
+        x = np.array([1, 2, 3])
+        y = np.array([4, 5, 6])
+
+        tab = Tabulation(x,y)
+
+        resampled = tab.resample(None)
+
+        self.assertTrue(np.all(resampled.x == x))
+        self.assertTrue(np.all(resampled.y == y))
+
+        # bandwidth_rms with dx=None
+        boxcar = Tabulation((0.,10.),(1.,1.))
+        value = 5
+
+        self.assertTrue(np.abs(boxcar.bandwidth_rms() - value) == 0.)
+
+        # Test multiplication of Two Tabulations
+        x1 = np.array([1, 2, 3])
+        y1 = np.array([4, 5, 6])
+        tab1 = Tabulation(x1, y1)
+
+        x2 = np.array([1, 2, 3])
+        y2 = np.array([1, 2, 3])
+        tab2 = Tabulation(x2, y2)
+
+        result = tab1 * tab2
+
+        expected_x = [1, 2, 3] # Merged x values
+
+        self.assertTrue(np.array_equal(result.x, expected_x))
+        self.assertTrue(np.array_equal(result(x), tab1(x) * tab2(x)))
+        self.assertTrue(np.array_equal(result.y, tab1.y * tab2.y))
```

Comment:
For completeness, should also check `result(x)` against `y1 * y2` to make sure that `tab1.y` and `tab2.y` are actually what you expect.